### DISCUSSION TITLE
Simple middleware example?

### DISCUSSION THREAD

**Discussion by ra0x3** [hyperium/hyper#2557] [Q&A] [answered]
### Issue Type

- [x] Question
- [ ] Bug
- [ ] Feature request

### Description

Are there any complete, simple middleware service examples out there (e.g., a simple request logger)? I've read the [middleware section of the Service docs here](
https://docs.rs/hyper/0.13.10/hyper/service/trait.Service.html) and my head is still spinning as to add a simple middleware service _then_ pass it to a request handler.

**Comment by seanmonstar** [comment, marked as answer]
There's [tower-http](https://github.com/tower-rs/tower-http). It's preparing its first major release, but browsing the code may be helpful. It's essentially a repository of useful HTTP middleware for `Service`.

**Reply by ra0x3** [reply]
Thanks @seanmonstar  (Apologies for creating an issue)
